src/app/components/Loop/QuoteModal.tsx

```diff
@@ -0,0 +1,264 @@
+    const req: GetLoopOutArguments = {
+      amt: this.props.amt,
+      dest: this.props.dest,
+      loop_out_channel: this.props.chan,
+      max_miner_fee: loopOutQuote.miner_fee,
+      max_prepay_amt: loopOutQuote.prepay_amt,
+      max_prepay_routing_fee: loopOutQuote.prepay_amt,
+      max_swap_fee: loopOutQuote.swap_fee,
+      max_swap_routing_fee: loopOutQuote.swap_fee,
+      sweep_conf_target: this.props.sct,
+    };
+    this.props.getLoopOut(req);
+    setTimeout(() => {
+      message.info(`Attempting ${this.props.type}`, 2);
+    }, 1000);
+    setTimeout(() => {
+      this.props.onClose();
+    }, 3000);
+  };
+
+  private loopIn = () => {
+    // get values from loopOutQuote for default loopOut
+    const loopInQuote = this.props.loopQuote;
+    const loopIn = this.props.loop;
+    if (loopInQuote === null) {
+      return null;
+    }
+    if (loopIn === null) {
+      return null;
+    }
+    const req: GetLoopInArguments = {
+      amt: this.props.amt,
+      loop_in_channel: this.props.chan,
+      max_miner_fee: loopInQuote.miner_fee,
+      max_swap_fee: loopInQuote.swap_fee,
+      external_htlc: this.props.htlc,
+    };
+
+    this.props.getLoopIn(req);
+    setTimeout(() => {
+      message.info(`Attempting ${this.props.type}`, 2);
+    }, 1000);
+    setTimeout(() => {
+      this.props.onClose();
+    }, 3000);
+  };
+
+  private advLoopOut = () => {
+    // get values from loopOutQuote for default loopOut
+    const loopOutQuote = this.props.loopQuote;
+    const loopOut = this.props.loop;
+    if (loopOutQuote === null) {
+      return null;
+    }
+    if (loopOut === null) {
+      return null;
+    }
+    const req: GetLoopOutArguments = {
+      amt: this.props.amt,
+      dest: this.props.dest,
+      loop_out_channel: this.props.chan,
+      max_miner_fee: this.props.mf,
+      max_prepay_amt: this.props.pre,
+      max_prepay_routing_fee: this.props.pre,
+      max_swap_fee: this.props.sf,
+      max_swap_routing_fee: this.props.srf,
+      sweep_conf_target: this.props.sct,
+    };
+    this.props.getLoopOut(req);
+    setTimeout(() => {
+      message.info(`Attempting advanced ${this.props.type}`, 2);
+    }, 1000);
+    setTimeout(() => {
+      this.props.onClose();
+    }, 3000);
+  };
+
+  private advLoopIn = () => {
+    // get values from loopOutQuote for default loopOut
+    const loopInQuote = this.props.loopQuote;
+    const loopIn = this.props.loop;
+    if (loopInQuote === null) {
+      return null;
+    }
+    if (loopIn === null) {
+      return null;
+    }
+    const req: GetLoopInArguments = {
+      amt: this.props.amt,
+      loop_in_channel: this.props.chan,
+      max_miner_fee: this.props.mf,
+      max_swap_fee: this.props.sf,
+      external_htlc: this.props.htlc,
+    };
+
+    this.props.getLoopIn(req);
+    setTimeout(() => {
+      message.info(`Attempting ${this.props.type}`, 2);
+    }, 1000);
+    setTimeout(() => {
+      this.props.onClose();
+    }, 3000);
+  };
+}
```

Comment:
I would consolidate the adv loop functions & buttons into the regular ones, and just form the `req` object based on if the `adv` prop is present or not.

Reply:
consolidated advanced loops with ternary